### DISCUSSION TITLE
Using Vercel CLI to get status of deployment

### DISCUSSION THREAD

**Discussion by adamstrawson** [vercel/community#3142] [Help] [answered]
### Summary

Prerequisites:
- We're using the Vercel Git Integration
- We have environment branches to imitate envs within Vercel (`dev`, `sandbox`, `main/production`)

I'd like to use the Vercel CLI to understand the status of the deploy to an environment (above), but struggling to find the correct way to achieve this. 

We run Github Actions which does the merge/rebase into the appropriate branch above, Vercel picks up the change to the branch via the Git Integration, and starts the deploy.  Within the GitHub Action, we'd like to verify that the deploy has succeeded.

The problem i'm having is that the deploymentAlias for the environment will always report back as `Ready`, because it's for the previous deployment. 

Lets assume the domain alias `portal.foo.com` is mapped to the branch `dev`

Running the command `vercel inspect https://portal-git-dev-foo.vercel.app` returns the below:

```
    name	portal
    status	● Ready
    url	https://portal-1ehhinsri-foo.vercel.app
```

I then merge a change into the branch `dev`, which triggers a build

Running the command `vercel inspect https://portal-git-dev-foo.vercel.app` returns the same output above, despite there being a queued build for the alias `https://portal-git-dev-foo.vercel.app`, both deployments have the same alias now as expected.

If I inspect the specific preview URL I get the expected result, however within Github actions I don't know the preview URL, only the alias URL. 

```
    name	portal
    status	● Queued
    url		https://portal-b7x74545-foo.vercel.app
```

Is there a way I can use the Vercel CLI to view the _latest_ deployment for an alias URL.

Alternatively, can I use `vercel ls <project>` and filter by branch using `-m/--meta` (I can't find any docs on what metadata is available here)? I can then use `awk` to pull out the status I need?

### Steps to Reproduce

N/A

**Comment by adamstrawson** [comment, marked as answer]
I've found the solution to my issue, specifically with using `-m/--meta`.

For anyone else coming across this discussion, I ended up needing to use the Vercel Rest API `https://api.vercel.com/v6/deployments` to view the full deployment response, this allowed me to view the `meta` data available. 

From this I was then able to use `vercel ls <project> -m githubCommitRef=dev` to return the correct deployments. 

In the end I actually ended up just using the API to do what we needed as the `json` response was easier to parse than the CLI output because of the noise that gets output (Might be a nice feature request to be able to change the output behaviour of the CLI, similar to GCP where you can use the `--format` arg.)

**Reply by claudiowilson** [reply]
@adamstrawson thank you so much for this! there was not a lot of documentation on the meta tag.

how did you filter via meta tag using the rest API? doesn't seem to be an arg for it

**Reply by claudiowilson** [reply]
also, for anyone looking for sample meta information you can filter by:

```
  "meta": {
    "githubCommitAuthorName": "First Last",
    "githubCommitMessage": "fix mobile width",
    "githubCommitOrg": "nickeltechnology",
    "githubCommitRef": "claudio/branch-name",
    "githubCommitRepo": "repo-name",
    "githubCommitSha": "bedb98b544c44d91e873ff227f1xfdce748f5563",
    "githubDeployment": "1",
    "githubOrg": "nickeltechnology",
    "githubRepo": "repo-name",
    "githubRepoOwnerType": "Organization",
    "githubCommitRepoId": "1231231",
    "githubRepoId": "123123",
    "githubRepoVisibility": "public",
    "githubCommitAuthorLogin": "claudiowilson",
    "branchAlias": "branchalias"
  },
```

**Reply by adamstrawson** [reply]
@claudiowilson The API doesn't allow filtering, so I ended up just pipping through `jq` 

Here's an example, which will return the deployment `state` for app `bar`, with the branch `dev` and commitSha `bedb98b544c44d91e873ff227f1xfdce748f5563` 

```bash
curl -X GET "https://api.vercel.com/v6/deployments?teamId=foo&app=bar" -H "Authorization: Bearer <snip>" --silent | jq -r '.deployments[] | select(.meta.githubCommitRef == "dev" and .meta.githubCommitSha == "bedb98b544c44d91e873ff227f1xfdce748f5563")  | .state'
```
